django_guid/middleware.py

```diff
@@ -14,10 +14,11 @@
 
 class GuidMiddleware(object):
     """
-    Checks for an existing GUID (correlation ID) in a request's headers.
-    If a header value is found, the value is validated as a GUID and stored, before the request is passed to the
-    next middleware.
-    If no value is found, or one is found but is invalid, we generate and store a new GUID on the thread.
+    Checks for a GUID (correlation ID) in an incoming request's headers and assigns it to the thread if found.
+    If none is found, one is generated.
+
+    As every request spawns a new thread, this makes it possible to add logger IDs that only span a single request.
```

Comment:
Not how it works, and not really needed to be explained in detail here. Each WSGI implementation works differently, but they all have a set of threads available and only use those.